Snakemake/workflow/rules/prepare_files_for_tsinfer.smk

```diff
@@ -1,115 +1,116 @@
     input:
-        vcf = rules.get_af.output.new_file,
-        #major = rules.get_major.output
-    output: f'{vcfdir}/{{chromosome}}_phased_info.vcf' # this is removing both the .gz and the decompressed file 
-    threads: 1
-    resources: cpus=1, mem_mb=4000, time_min=5
-    log: 'logs/decompress_{chromosome}.log'
-    shell:
+        ancestral_file
+    output:
+        temp(f'{vcfdir}/Ancestral{{chromosome}}.txt')
+    params:
+        chrNum = lambda wc: wc.get('chromosome')[3:]
+    log: 'logs/Extract_ancestral_chromosome_{chromosome}.log'
+    resources: cpus=1, mem_mb=32000, time_min=30
+    shell:  
         """
-        gunzip {input.vcf}
+        grep "{params.chrNum}_" {input} | grep -xv 'ambiguous' > {output}
         """
 
-rule extract_vcf_pos:
-    input: rules.get_major.output
-        #rules.decompress.output
+rule join_major_ancestral:
+    input:
+        ancestral = rules.extract_ancestral_chromosome.output,
+        major = rules.get_major.output
     output:
-        file = temp(f'{vcfdir}/VcfPos{{chromosome}}.txt'),
-        #sites = temp('{chromosome}_sites.list')  
-    conda: "bcftools"
-    threads: 1
-    resources: cpus=1, mem_mb=4000, time_min=5
-    log: 'logs/extract_vcf_pos_{chromosome}.log'
+        temp(f'{vcfdir}/MajAAnc_{{chromosome}}.txt')
+    log: 'logs/Join_major_ancestral_{chromosome}.log'
+    resources: cpus=1, mem_mb=32000, time_min=60
     shell:
         """
-        cut -d',' -f 1 {input} > {output}
+        awk -F"," 'NR==FNR{{A[$1]=$2;next}}{{print$0 FS (A[$1]?A[$1]:"0")}}' {input.ancestral} {input.major} > {output}
         """
-    # why not just take the first col of the MajorAlle table??
-    # # 
-    # bcftools query -f '%CHROM %POS\n' {input} > {output.sites}
-    #     awk '{{print $1"_"$2}}' {output.sites} > {output.file}
 
-if config['ancestralAllele'] not null:
-    ancestral_file = config['ancestralAllele']
-else
-   ancestral_file = "AncestralAllele/AncestralAllele_Vcf.txt"
-
-rule match_ancestral_vcf:
+rule determine_ancestral_major:
     input:
-        vcfPos = rules.extract_vcf_pos.output.file,
-        ancestral = ancestral_file
-        #ancestral = "AncestralAllele/AncestralAllele_Vcf.txt",
-        major = rules.get_major.output
-    output: 
-        file = temp(f'{vcfdir}/AncestralVcfMatch{{chromosome}}.txt'),
-    params:
-        chrNum = lambda wc: wc.get('chromosome')[3:],
-        ancestral_sites = f'{vcfdir}/{{chromosome}}.aa',
-        tmp_file = f'{vcfdir}/{{chromosome}}.tmp',
-    log: 'logs/match_ancestral_vcf_{chromosome}.log'
+        rules.join_major_ancestral.output
+    output:
+        temp(f'{vcfdir}/MajOAnc_{{chromosome}}.txt')
+    log: 'logs/Determine_major_ancestral_{chromosome}.log'
     shell:
         """
-        grep "{params.chrNum}_" {input.ancestral} | grep -xv 'ambiguous' > {params.ancestral_sites}
-        echo done
+        awk -F","  '{{if ($3 == 0) {{print $1" "$2}} else {{print $1" "$3}}}}' {input} > {output}
```

Comment:
This is now done is a way that it extracts ancestral, extracts major, merges then together and prints the corresponding column - I think that will be quicker than the grep loop